### PR TITLE
feat(value): set function

### LINKED ISSUE #2: feat(value): be able to set function into object

### PR DESCRIPTION
**Pull request by tony-go**
Fix #2

### CODE REVIEW

File: src/engine/include/sourcemeta/includejs/engine_value.h [hunk 1]
```diff
@@ -24,6 +25,9 @@ class SOURCEMETA_INCLUDEJS_ENGINE_EXPORT Value {
   Value(Value &&other) noexcept;
   Value(const Value &other) = delete;
 
+  using Function = std::function<Value(std::span<Value> arguments)>;
+  using FunctionStorage = std::unordered_map<void *, Value::Function>;
```
Comment (jviotti) [inline src/engine/include/sourcemeta/includejs/engine_value.h:29]: I don't think `Value::Function` is valid. I think you want just `Function`?

Comment (tony-go) [reply src/engine/include/sourcemeta/includejs/engine_value.h:29]: Both don't work. I'll check again.

Comment (tony-go) [reply src/engine/include/sourcemeta/includejs/engine_value.h:29]: For instance I have the same Error with `std::span` that is properly include at the top of the file haha

Comment (tony-go) [reply src/engine/include/sourcemeta/includejs/engine_value.h:29]:
```
/Users/tonygorez/oss/includejs/src/engine/include/sourcemeta/includejs/engine_value.h:28:45: error: no member named 'span' in namespace 'std'
  using Function = std::function<Value(std::span<Value> arguments)>;
                                       ~~~~~^
```

Comment (tony-go) [reply src/engine/include/sourcemeta/includejs/engine_value.h:29]: All worked nicely there: https://github.com/sourcemeta/includejs/compare/set-function?expand=1

I wonder where I fucked up I'll investigate more

**Review by jviotti — APPROVED** [review]

**Pull request merged**